src/charm.py

```diff
@@ -127,8 +137,19 @@ def _configure(self, _):
             container.restart(self._name)
             logger.info("Prometheus (re)started")
 
+        # Provide connection info to any remote write clients.
+        self._set_remote_write()
```

Comment:
The name of this function is a bit misleading. As I understand you are "setting" remote write (or enabling remote write) though the "self._command()" method by adding an additional command line option. This function is providing and address for the remote write client. If you agree wouldn't this method be better named as `self._set_remote_write_address()` or something like that. 

Also why does this need to happen in the charm ? I fail to see why the library can not do this and thereby make it transparent to the charm. In fact I would have expected that the library sets the remote write address in response to a relation joined with the remote write client.

Reply:
Don’t forget the docu in README. That limitation hits juju admins way more than charm authors

Reply:
First paragraph: I agree completely
Second paragraph: I will deliberate

Reply:
Given discussion on the way we use ingress with remote write in prometheus, I think it makes sense to leave this up to the charm. The implementation here is very use case specific and I think two charms using this library are not particularly likely to use the same logic. Thoughts @balbirthomas?

Reply:
Your call @dstathis . Personally I would ensure as much as possible the entire responsibility of managing the relation is within the library. This way you docs can proudly say "Hey look using this library is easy - just instantiate it passing these constructor arguments, and go for a picnic" :-).

Reply:
My concern here is that the default behavior will be relevant to only prometheus-k8s/grafana-agent-k8s and anyone using the library will have to define this anyway. For example the docs will have to read "It's really easy unless you want a different behavior regarding ingress, then you need to pass a custom method to the constructor".

Reply:
1. Even if the default behaviour is relevant only to grafana agent the fact is AFAIK grafana agent is the dominant use case for this library. 
2. By putting it in the library you get the opportunity to also document it in **one and the same place**, how address selection process works by default and how rest of the library woks. I view this as another reason to put it into the library. Of course you can mention in passing that if this default behaviour is not desired then the library object can be derived and extended.
3. Having all the behaviour of the remote write interface located in **one and the same place** will be the more maintainable solution in the long run.